### PR TITLE
Use public contact ID in Model\Profile::getVCardHtml instead of potentially self-contact ID

### LINKED ISSUE #13603: Network posts button leads to error page
- [x] I have searched open and closed issues for duplicates

### Bug Description

Clicking on the Network posts button leads to an error. 

```
Exception thrown in /var/www/html/friendica/src/Module/Contact/Conversations.php:84
Stack trace:
#0 /var/www/html/friendica/src/BaseModule.php(250): Friendica\Module\Contact\Conversations->content()
#1 /var/www/html/friendica/src/App.php(711): Friendica\BaseModule->run()
#2 /var/www/html/friendica/index.php(55): Friendica\App->runFrontend()
#3 {main}
```

### Screenshots

Profile page of a user with the Network posts button

Result after clicking on the Network posts button

### Platform Info

Friendica Version: 2023.09-rc

Friendica Source: git

PHP version: 8.1

SQL version: 10.5

### PR DESCRIPTION
**Pull request by MrPetovan**
Fix #13603

### CODE REVIEW

File: src/Model/Profile.php [hunk 1]
```diff
@@ -307,7 +307,7 @@ public static function getVCardHtml(array $profile, bool $block, bool $show_cont
 
 		$profile_url = $profile['url'];
 
-		$cid = $profile['id'];
+		$cid = Contact::getIdForURL($profile['url']);
```
Comment (annando) [inline src/Model/Profile.php]: You could use `getPublicAndUserContactID` instead.

You could also use the `uri-id` field of the profile. With this you can query the public contact.

Comment (MrPetovan) [reply src/Model/Profile.php]: Please just tell me which way is the least costly.

Comment (annando) [reply src/Model/Profile.php]: You could use 
```
$contact = Contact::selectFirst(['uid*], ['uri-id` => $profile['uri-id], 'uid' => 0]};
```

Comment (MrPetovan) [reply src/Model/Profile.php]: Thanks!

**Pull request merged**